src/prisma/schedule.ts

```diff
@@ -37,3 +39,93 @@ async function initiateSchedule(scheduleyear: number) {
   });
   return schedule;
 }
+
+async function updateCurrentSchedule(
+  scheduleId: any,
+  courses: CourseSectionInput[]
+) {
+  // Simply creates a new schedule instead of updating the current schedule
+
+  let currentSchedule;
+
+  try {
+    currentSchedule = await prisma.schedule.findFirst({
+      where: {
+        id: parseInt(scheduleId),
+      },
+    });
+  } catch (error) {
+    return {
+      success: false,
+      message: "Couldn't find schedule",
+      errors: error,
+    };
+  }
```

Comment:
Can this be removed if we're going to by default create an new instance?

Reply:
Sure, I can remove it and just use the current year as the input for year in the new schedule

Reply:
On second thought I think it makes sense given the endpoint specifically states this behaviour...

If the `id` is passed in, you'd need to delete all the `CourseSection` instances associated with that schedule ID then insert like you're doing right now.

I think you will need to make it a transaction so it only succeeds if the entire operation works?

Reply:
Sounds good

Reply:
Made the suggested changes @aomi